client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -231,9 +233,14 @@ export class PlanFeatures2023Grid extends Component<
 	PlanFeatures2023GridType,
 	PlanFeatures2023GridState
 > {
-	state = {
-		showPlansComparisonGrid: false,
-	};
+	constructor( props: PlanFeatures2023GridType ) {
+		super( props );
+
+		this.state = {
+			showPlansComparisonGrid: false,
+			selectedStorage: {},
+		};
+	}
```

Comment:
I think we don't need explicit constructor call now right? just

```
state = {
    showPlansComparisonGrid: false,
    selectedStorage: {},
};

Reply:
I was mentioning this in a recent response. There's something about moving the state into a constructor that resolves a typescript error that appears in `renderStorageAddOnDropdown`

<img width="1260" alt="Screenshot 2023-07-11 at 9 53 21 AM" src="https://github.com/Automattic/wp-calypso/assets/5414230/e85e13ac-13c3-4016-9896-634ea3348681">

https://github.com/Automattic/wp-calypso/pull/79041#discussion_r1258938572

Again, I don't fully understand the why, but I figured that switching to the constructor isn't too big of an issue

Reply:
Either way should be fine. These would move to `useState`, when/if we get to that refactor.

Reply:
Props to @jdc91 for tracking down why this TS error was being thrown 🎉 

p1689110327073639/1689107860.312579-slack-CV2TX2PAN